src/Form/Theme.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * Handle Theme registration
+ *
+ * @package Give
+ * @since 2.7.0
+ */
+
+namespace Give\Form;
+
+use function Give\Form\Theme\getActiveID;
+use function Give\Form\Theme\get;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Theme class.
+ *
+ * @since 2.7.0
+ */
+class Theme {
+	/**
+	 * Theme data
+	 *
+	 * @var array
+	 */
+	private $data;
+
+	/**
+	 * RegisterTheme constructor.
+	 *
+	 * @param array $args    {
+	 *
+	 * @type string $id      Theme ID
+	 * @type string $name    Theme name
+	 * @type string $image   Theme image
+	 * @type string $title   Theme title (optional). Can be contain whitelisted HTML tags: strong, a.
+	 * @type array  $options Array representation of setting.
+	 *
+	 * }
+	 */
+	public function __construct( $args ) {
+		$this->data = $args;
```

Comment:
Please break `$args` up into individual parameters and then have one property per parameter. Having a single `$this->data` class property tells you nothing about the shape of the class. Your getters should be as simple as `return $this->id`.

Reply:
@JasonTheAdams will it be fine if I will create property dynamically?

```
foreach( $args as $property as $value ){
    $this->{$property} = $value
}
```

Reply:
Why? What's the benefit? Why not just pass property as a parameter?

Reply:
@JasonTheAdams  are you saying you pass individual to constructor because currently, I have five properties which can be increased in the future. Will it look good to pass lots of params to function?

```
 * @param array $args    {
	 *
	 * @type string $id      Theme ID
	 * @type string $name    Theme name
	 * @type string $image   Theme image
	 * @type string $title   Theme title (optional). Can be contain whitelisted HTML tags: strong, a.
	 * @type array  $options Array representation of setting.
	 *
	 * }
```

Reply:
I would say 5-7 is the high but acceptable range. Otherwise, have a whitelist of acceptable arguments and check against those.

Reply:
this will be resolve in https://github.com/impress-org/givewp/pull/4518